### ISSUE TITLE
Allow Utilization of Original Training Data during Backtesting

### ISSUE THREAD

**Issue by Nicholas-Amirsoleimani** [unit8co/darts#1859]
**Is your feature request related to a current problem? Please describe.**
Yes, the current backtesting implementation in Darts seems to only leverage the provided 'series' (test set) when retraining the model for rolling forecasts. This can be limiting in scenarios where one would want to use a combination of the original training data (stored as model.training_series) and the latest observations from the test set to retrain the model.

**Describe proposed solution**
To solve this issue, an additional parameter could be added to the backtest method, let's call it use_original_train_data. If set to True, the method should use the stored model.training_series and the latest observations from the test set (based on the train_length parameter) for retraining the model during the rolling forecast.

**Describe potential alternatives**
An alternative solution could be to merge the train and test series into one, prior to calling the backtest method. However, this alternative approach is less explicit and might cause confusion since it alters the clear separation between training and testing data.

**Additional context**
This feature can provide additional flexibility when using Darts for time series forecasting tasks, as it enables users to exploit all available historical data when performing rolling forecasts. This can be particularly useful in scenarios where recent observations do not carry all relevant information for future forecasts, and a larger historical context from the training data is necessary.

**Comment by madtoinou** [comment]
Hi @Nicholas-Amirsoleimani,

If you obtained the train and test series using `split_after/before()`, the original "full" series should already be available or could easily be reconstructed with `append()` ([doc](https://unit8co.github.io/darts/generated_api/darts.timeseries.html?highlight=stack#darts.timeseries.TimeSeries.append)) and be passed to the `series` argument of `backtest()`.

Moreover, for global models trained on several series, the `training_series` attribute will be `None`, causing problems with the approach you described.

I would be rather against introducing a new argument as a simple operation on the `TimeSeries` could solve the problem.

**Comment by Nicholas-Amirsoleimani** [comment]
You're completely right, I didn't think to put the entire dataset in. Thanks!
